src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -17,4 +19,11 @@ class LeonardoService(gdDAO: DataprocDAO, dbRef: DbReference)(implicit val execu
     }
   }
 
+  def deleteCluster(googleProject: GoogleProject, clusterName: String): Future[Int] = {
```

Comment:
TOL: I'm wondering if deleting in Google and deleting in the Leo DB should be done independently. In this code, we delete in Leo DB _only after_ we successfully delete in Google. Could there be a situation where the cluster does not exist in Google (for whatever reason), and we still want to delete it in the DB? I assume if the user hits the delete endpoint they'd want us to delete as much as possible, even if things are in a bad state. Thoughts?

If we want to do the deletes independently, could do it like:
```
val googleDeleteFuture = gdDAO.deleteCluster(googleProject, clusterName)
val dbDeleteFuture = dbRef.inTransaction { _.clusterQuery.deleteCluster(googleProject, clusterName) }
Future.sequence(Seq(googleDeleteFuture, dbDeleteFuture)).mapTo[Unit]
```

Reply:
I think that this should:

1. query the db for the cluster
2. if status != Deleting or Deleted, tell Google to delete it
3. if status != Deleting or Deleted, flip the status to Deleting
4. (later, and depending on implementation) send a message to deletion poller to get lookin'